File: MapA7/MapA7/Model/FileTable.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model
{
    public class FileTable : IFileTable
    {
        private Dictionary<int, Tuple<String, StreamReader>> dict;
        private int fd = 0;

        public FileTable()
        {
            dict = new Dictionary<int, Tuple<String, StreamReader>>();
        }

        public int Add(Tuple<String, StreamReader> item)
        {
            fd++;
            dict.Add(fd, item);
            return fd;
        }

        public bool Exists(String filename)
        {
            foreach (Tuple<String, StreamReader> item in dict.Values)
                if (filename == item.Item1) return true;
            return false;
        }

        public void Remove(int fd)
        {
            dict.Remove(fd);
        }

        public StreamReader GetValue(int fd)
        {
            if (dict.ContainsKey(fd))
                return dict[fd].Item2;
            return null;
        }

        public void Clear()
        {
            dict.Clear();
        }

        public String Tostring()
        {
            String str = "FileTable = {";
            String first = "";
            foreach (var item in dict)
            {
                str += first;
                str += item.Key + ": " + item.Value.Item1;
                first = ", ";
            }
            str += "}";
            return str;

        }
    }
}

File: MapGUI3/src/Model/FileTable.java
package Model;

import Exceptions.CustomException;

import java.io.BufferedReader;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;

public class FileTable implements IFileTable {
    private Map<Integer,Map.Entry<String, BufferedReader>> dictionar;
    private Integer key;

    public FileTable() {
        dictionar = new HashMap<>();
        key=1;
    }
    public boolean exists(String filename){
        for (Map.Entry<String, BufferedReader> tuple: dictionar.values())
            if (filename.equals(tuple.getKey())) return true;
        return false;
    }
    public Map.Entry<String, BufferedReader> get(int key){
        return dictionar.get(key);
    }
    public Set<Map.Entry<Integer,Map.Entry<String,BufferedReader>>> entrySet(){
        return dictionar.entrySet();
    }
    public int put(Map.Entry<String, BufferedReader> pair){ dictionar.put(key,pair); key=key+1; return key-1;}
    public void remove(int fileDescriptor){dictionar.remove(fileDescriptor);}
    public BufferedReader getBufferReader(int fileDescriptor) {
        if(!dictionar.containsKey(fileDescriptor))
            throw new CustomException("Invalid file descriptor\n");
        else
            return dictionar.get(fileDescriptor).getValue();
    }
    public Map<Integer,Map.Entry<String, BufferedReader>> getContent() {return dictionar;}
    public void clear(){dictionar.clear();}
    public String toString(){return "FileTable = " + dictionar.toString();}
}

